server.go

```diff
@@ -386,6 +386,46 @@ func (server *Server) createTLSConfig(entryPointName string, tlsOption *TLS, rou
 	// BuildNameToCertificate parses the CommonName and SubjectAlternateName fields
 	// in each certificate and populates the config.NameToCertificate map.
 	config.BuildNameToCertificate()
+	//Set the minimum TLS version if set in the config TOML
+	if server.globalConfiguration.EntryPoints[entryPointName].TLS.MinVersion != "" {
+		config.PreferServerCipherSuites = true
+		switch server.globalConfiguration.EntryPoints[entryPointName].TLS.MinVersion {
+		case "VersionTLS10":
+			config.MinVersion = tls.VersionTLS10
+		case "VersionTLS11":
+			config.MinVersion = tls.VersionTLS11
+		case "VersionTLS12":
+			config.MinVersion = tls.VersionTLS12
+		}
+	}
+	//Set the list of CipherSuites if set in the config TOML
+	if server.globalConfiguration.EntryPoints[entryPointName].TLS.CipherSuites != nil {
+		config.CipherSuites = make([]uint16, 0)
+		for _, cipher := range server.globalConfiguration.EntryPoints[entryPointName].TLS.CipherSuites {
+			switch cipher {
```

Comment:
Well I'm conflicted between setting the ciphersuites map in the TLS struct definition, or as a global. Don't really want to set it in the entrypoint code (in `server.go`). Would be nice if we could do something like `TLS.CipherSuites.Enabled` and `TLS.CipherSuites.Available`. That way I could set the `Available` map in the struct definition. Could verify using:

if TLS.Ciphersuites.Available[cipher] AND TLS.Ciphersuites.Enabled[cipher] exists

That way if we removed a cipher suite from the codebase, we wouldn't allow users to add  one in. It would have to exist in our map, and in the config file.

Thoughts?